crates/bitwarden/src/platform/sync.rs

```diff
@@ -80,20 +90,35 @@ impl SyncResponse {
 
         Ok(SyncResponse {
             profile: ProfileResponse::process_response(profile, enc)?,
+            folders: response
```

Comment:
This function is doing the map, into_iter, try_into, collect dance a lot, we can refactor that to a function like this, what do you think?

```rust
        fn try_into_iter<In, InItem, Out, OutItem>(iter: In) -> Result<Out, InItem::Error>
        where
            In: IntoIterator<Item = InItem>,
            InItem: TryInto<OutItem>,
            Out: FromIterator<OutItem>,
        {
            iter.into_iter().map(|i| i.try_into()).collect()
        }

        Ok(SyncResponse {
            profile: ProfileResponse::process_response(profile, enc)?,
            folders: try_into_iter(response.folders.ok_or(Error::MissingFields)?)?,
            collections: try_into_iter(response.collections.ok_or(Error::MissingFields)?)?,
            ciphers: try_into_iter(ciphers)?,
            domains: response.domains.map(|d| (*d).try_into()).transpose()?,
            policies: try_into_iter(response.policies.ok_or(Error::MissingFields)?)?,
            sends: try_into_iter(response.sends.ok_or(Error::MissingFields)?)?,
        })
```

Reply:
I like it.